cornice/validators/_colander.py

```diff
@@ -32,12 +32,11 @@ def body_validator(request, schema=None, deserializer=None, **kwargs):
     class RequestSchema(colander.MappingSchema):
         body = _ensure_instantiated(schema)
 
-        def deserialize(self, cstruct=colander.null):
-            appstruct = super(RequestSchema, self).deserialize(cstruct)
-            return appstruct['body']
-
     validator(request, RequestSchema(), deserializer, **kwargs)
 
+    if 'body' in request.validated:
```

Comment:
I didn't get that part :( When is it relevant to have this test? With the above RequestSchema it looks like `body` is mandatory no?